src/Microsoft.Identity.Web.OWIN/AppBuilderExtension.cs

```diff
@@ -175,6 +175,23 @@ public static IAppBuilder AddMicrosoftIdentityWebApp(
 
                                 if (clientInfoFromServer != null && clientInfoFromServer.UniqueTenantIdentifier != null && clientInfoFromServer.UniqueObjectIdentifier != null)
                                 {
+                                    var uniqueTenantIdentifierClaim = context.AuthenticationTicket.Identity.FindFirst(c => c.Type == ClaimConstants.UniqueTenantIdentifier);
+                                    var uniqueObjectIdentifierClaim = context.AuthenticationTicket.Identity.FindFirst(c => c.Type == ClaimConstants.UniqueObjectIdentifier);
+                                    if (uniqueTenantIdentifierClaim != null && !string.Equals(clientInfoFromServer.UniqueTenantIdentifier, uniqueTenantIdentifierClaim.Value, StringComparison.OrdinalIgnoreCase))
+                                    {
+                                        throw new InternalClaimDetectedException($"The claim \"{ClaimConstants.UniqueTenantIdentifier}\" is reserved for internal use by this library. To ensure proper functionality and avoid conflicts, please remove or rename this claim in your ID Token.")
```

Comment:
Can you extract this error message to where other error messages live? Maybe you can include it in the definition for `InternalClaimsDetectedException` and just pass the claim name as an argument.